File: backends/ktor/src/main/kotlin/com/hellotimepro/ktor/repository/UserRepository.kt
package com.hellotimepro.ktor.repository

import com.hellotimepro.ktor.db.Users
import com.hellotimepro.ktor.domain.User
import org.jetbrains.exposed.sql.*
import java.util.UUID

/** 假定调用时已处于一个 Exposed 事务中（由 service 的 dbQuery 开启）。 */
class UserRepository {
    fun findById(id: UUID): User? =
        Users.select { Users.id eq id }.firstOrNull()?.toUser()

    fun findByEmail(email: String): User? =
        Users.select { Users.email eq email }.firstOrNull()?.toUser()

    fun existsByEmail(email: String): Boolean =
        Users.select { Users.email eq email }.count() > 0L

    fun existsByNickname(nickname: String): Boolean =
        Users.select { Users.nickname eq nickname }.count() > 0L

    fun findAllByIds(ids: Collection<UUID>): List<User> =
        if (ids.isEmpty()) emptyList()
        else Users.select { Users.id inList ids }.map { it.toUser() }

    fun insert(user: User) {
        Users.insert {
            it[id] = user.id
            it[email] = user.email
            it[passwordHash] = user.passwordHash
            it[nickname] = user.nickname
            it[avatarId] = user.avatarId
            it[createdAt] = user.createdAt
            it[updatedAt] = user.updatedAt
        }
    }

    fun update(user: User) {
        Users.update({ Users.id eq user.id }) {
            it[email] = user.email
            it[passwordHash] = user.passwordHash
            it[nickname] = user.nickname
            it[avatarId] = user.avatarId
            it[updatedAt] = user.updatedAt
        }
    }
}

File: backends/aspnet/src/Repositories/UserRepository.cs
using HelloTimePro.Aspnet.Domain;
using HelloTimePro.Aspnet.Infrastructure;
using Microsoft.EntityFrameworkCore;

namespace HelloTimePro.Aspnet.Repositories;

/// <summary>
/// 用户仓库。DbContext 即工作单元：写方法只暂存变更，由 service 在业务边界统一 SaveChanges。
/// （查询默认 tracking，便于 service 直接修改已加载实体后保存。）
/// </summary>
public sealed class UserRepository
{
    private readonly AppDbContext _db;
    public UserRepository(AppDbContext db) => _db = db;

    public Task<User?> FindById(Guid id) =>
        _db.Users.FirstOrDefaultAsync(u => u.Id == id);

    public Task<User?> FindByEmail(string email) =>
        _db.Users.FirstOrDefaultAsync(u => u.Email == email);

    public Task<bool> ExistsByEmail(string email) =>
        _db.Users.AnyAsync(u => u.Email == email);

    public Task<bool> ExistsByNickname(string nickname) =>
        _db.Users.AnyAsync(u => u.Nickname == nickname);

    public async Task<List<User>> FindAllByIds(IReadOnlyCollection<Guid> ids) =>
        ids.Count == 0 ? new List<User>() : await _db.Users.Where(u => ids.Contains(u.Id)).ToListAsync();

    /// <summary>暂存新增（不保存）。</summary>
    public void Add(User user) => _db.Users.Add(user);
}
